tests/unit/test_spin.cpp

```diff
@@ -655,6 +657,18 @@ SECTION("Swap bra kets") {
   }
 }
 
+SECTION("Closed-shell spintrace CCD") {
+  // Energy expression
+  {
+    const auto input = ex<Sum>(ExprPtrList{parse_expr(
+        L"1/4 g{i_1,i_2;a_1,a_2} t{a_1,a_2;i_1,i_2}", Symmetry::antisymm)});
+    auto result = closed_shell_CC_spintrace(input);
+    REQUIRE(*result == *parse_expr(L"-g{i_1,i_2;a_1,a_2} t{a_1,a_2;i_2,i_1} + "
```

Comment:
this check seems to be non-portable, e.g. see https://github.com/ValeevGroup/SeQuant/actions/runs/6811607068/job/18522290446?pr=136#step:10:291

Reply:
Yeah the ordering of the addends in the result seems to be in reverse order on the CI compared to my local setup. No idea why that would happen, but I have added an explicit ordering step to the test case and now it succeeds.

Reply:
Canonicalizer output unfortunately depends on the Boost.Hash version and they changed how hashing works in 1.81 ... I don't want to reimplement hashing (and Boost.Hash > Std.Hash) so for some unit tests we have to provide both variants